models/user_input.js

```diff
@@ -0,0 +1,174 @@
+                            'description, timestamp, created_at, updated_at) ' +
+                            'VALUES ($1, $2, $3, $4, $5, $6, $6) returning id',
+                            [data.experiment_id, data.device_id, data.user_id,
+                            data.description, data.timestamp, time]);
+                    })
+                    .then((result) => {
+                        resolve(result.rows[0]);
+                    })
+                    .catch((err) => {
+                        console.log(err);
+                        reject(err);
+                    });
+            });
+        },
+
+        delete: (data) => {
+            var time = new Date().getTime();
+            return new Promise((resolve, reject) => {
+                _db.query('UPDATE user_inputs SET deleted_at = $2 WHERE id = $1 returning id', [data.id, time])
+                    .then((result) => {
+                        resolve(result.rows[0]);
+                    })
+                    .catch((err) => {
+                        reject(err);
+                    });
+            });
+        },
+
+        updateDescription: (data) => {
+            var time = new Date().getTime();
+            return new Promise((resolve, reject) => { 
+                _db.query('UPDATE user_inputs SET description = $2, updated_at = $3 WHERE id = $1 and deleted_at = 0 returning description',
+                    [data.id, data.description, time])
+                    .then((result) => {
+                        resolve(result.rows[0]);
+                    })
+                    .catch((err) => {
+                        console.log(err);
+                        reject(err);
+                    });
+            });
+        },
+    }
+};
+
+function findOneById(id) {
+    return new Promise((resolve, reject) => {
+      _db.query('SELECT * FROM user_inputs WHERE id = $1 and deleted_at = 0', [id])
+        .then((result) => {
+          if (result.rows[0]) {
+            resolve(result.rows[0]);
+          }
+          else {
+            reject('no user input found')
+          }
+        })
+        .catch((err) => {
+          reject(err);
+        });
+    });
+}
+
+function validateUserInputData(data) {
+    return new Promise((resolve, reject) => {
+        if (!data.description) {
+            reject('invalid input: description missing')
+        }
+        else{
+            columns = ['user_id','device_id', 'description', 'experiment_id']
+            _Validator.validateColumns(data, columns)
+            .then(function () {
+                return _Validator.validateUserId(data.device_id)
+            })
+            .then(function () {
+                return _Validator.validateUserId(data.user_id)
+            })
+            .then(function() {
+                return _Validator.validateExperimentId(data.experiment_id)
+            })
+            .then((description) => {
+                data.description = description;
```

Comment:
What are you doing this for?

Reply:
I responded to your earlier comment

Reply:
What is the goal of this line? I'm not following

Reply:
Oh... now I see what you mean. Sorry, I had a moment. Fixing now

Reply:
Yay cool!